src/main/java/org/openrewrite/java/migrate/lang/UseVarKeyword.java

```diff
@@ -0,0 +1,162 @@
+            private final JavaType.Primitive BYTE_TYPE = JavaType.Primitive.Byte;
+            private final JavaTemplate template = JavaTemplate.builder(this::getCursor, "var #{} = #{any()}")
+                    .javaParser(JavaParser.fromJavaVersion()).build();
+
+            @Override
+            public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext executionContext) {
+                J.VariableDeclarations vd = (J.VariableDeclarations) super.visitVariableDeclarations(multiVariable, executionContext);
+
+                boolean isOutsideMethode = !determineIfIsInsideMethode(this.getCursor());
+                boolean isMethodeParameter = determineIfMethodeParameter(vd, this.getCursor());
+                if (isOutsideMethode || isMethodeParameter) return vd;
+
+                TypeTree typeExpression = vd.getTypeExpression();
+                boolean isByteVariable = typeExpression instanceof J.Primitive && BYTE_TYPE.equals(typeExpression.getType());
+                boolean isShortVariable = typeExpression instanceof J.Primitive && SHORT_TYPE.equals(typeExpression.getType());
+                if (isByteVariable || isShortVariable) return vd;
+
+                boolean definesNoVariable = vd.getVariables().isEmpty();
+                boolean isCompoundDefinition = vd.getVariables().size() < 1;
+                boolean isPureAssigment = isNull(vd.getTypeExpression());
+                if (definesNoVariable || isCompoundDefinition || isPureAssigment) return vd;
+
+                Expression initializer = vd.getVariables().get(0).getInitializer();
+                boolean isDeclarationOnly = isNull(initializer);
+                boolean isNullAssigment = initializer instanceof J.Literal && isNull(((J.Literal) initializer).getValue());
+                boolean alreadyUseVar = typeExpression instanceof J.Identifier && "var".equals(((J.Identifier) typeExpression).getSimpleName());
+                boolean isGeneric = typeExpression instanceof J.ParameterizedType; // todo implement generics!
+                if (alreadyUseVar || isDeclarationOnly || isNullAssigment || isGeneric) return vd;
+
+                J.VariableDeclarations result = transformToVar(vd);
+                return result;
+            }
+
+            private boolean determineIfMethodeParameter(@NotNull J.VariableDeclarations vd, @NotNull Cursor cursor) {
+                J.MethodDeclaration methodDeclaration = cursor.firstEnclosing(J.MethodDeclaration.class);
+                return nonNull(methodDeclaration) && methodDeclaration.getParameters().contains(vd);
+            }
+
+            /**
+             * Determines if a cursor is contained inside a Methode declaration without an intermediate Class declaration
+             * @param cursor value to determine
+             */
+            private boolean determineIfIsInsideMethode(@NotNull Cursor cursor) {
```

Comment:
I noticed that this excludes static and non-static initializers as in:
```java
class Foo {
    {
        Object o = new Object();
    }
    static {
        Object o = new Object();
    }
}
```
But feel free to supply a follow-up PR for that.

Reply:
Thanks for the review. 

>     * I found a problem with compound definitions, which currently would do the wrong thing.
_compound_ is the wrong term, multiple variable definition would be bette and should be covered by `UseVarKeywordTest#GeneralNotApplicable#simpleAssigment`.
According to [Java SE 8 Language Spec 15.26.2|https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.26.2] compound assignments are reassignments and therefore not a target for local variable type inference.

>     * I also noticed that it would currently not apply to code inside static and non-static initializers
>     * It might be cool to also support the variables of enhanced for-loops (e.g. `for (var s : List.of("a", "b"))`)
Yes they are not supported, yet. 
As I think about cool stuff, not covered stuff:
* initialization with ternary operator
* initialization via method invocations

Would I be an acceptable way to implement tests as marker and postpone the implementation to a followup?